peps/pep-0780.rst

```diff
@@ -0,0 +1,469 @@
+  maintainers, has stated
+  <https://github.com/python-cffi/cffi/pull/143#issuecomment-2580781899>`__
+  that it may be a good idea to fork ``cffi``, implement support for
+  free-threading, and only come back to the CFFI project with a single large PR
+  that adds support after the functionality "is reasonably well-tested (either
+  as tests or, better in this case, tested by being in use in various other
+  projects)". There are a lot of projects that depend on ``cffi``. They are
+  likely fine to start depending on a fork for free-threading only, however
+  depending on a fork for >=3.13 or for all Python versions seems like a much
+  larger ask, and more disruptive for distribution packagers.
+
+While these concrete examples may be addressed later this year by Cython and
+CFFI making compatible releases, the same issue is going to repeat further up
+the stack. The free-threading rollout is expected to take several years, and an
+environment marker for free-threading will make that rollout significantly less
+bumpy.
+
+Another important ABI feature that is not yet covered by environment markers is
+the bitness of the interpreter. In most cases, the ``sys_platform`` or
+``platform_system`` markers are enough, because there's only a single bitness
+in use per platform. This is not the case on Windows however: both 32-bit and
+64-bit Python interpreters are widely used on x86-64 Windows. Not being able to
+distinguish between the two may be relevant for packages that provide C
+extensions. For example, SciPy does not provide ``win32`` wheels (it isn't able
+to due to the lack of a suitable 32-bit compiler toolchain with Fortran
+support). Those wheels lacking can be awkward especially for projects where
+SciPy is an optional dependency only. In that case, it would be useful to be
+able to specify that SciPy is required *unless* the interpreter is a 32-bit one
+on Windows (c.f. `Require SciPy Unless on 32-bit win32`_) - it avoids users who
+install such packages to get failed from-source installs because of the missing
+wheels.
+
+
+Rationale
+=========
+
+The intention of this PEP is to introduce its core features with minimal impact
+on the existing ecosystem.
+The existing grammar proposed in :pep:`508` lends itself to a straightforward
+extension to include the new environment marker.
+
+Adding ``sys.abi_features`` to the Python Standard Library
+----------------------------------------------------------
+
+The Python standard library already has one mechanism to provide information
+about the ABI of the interpreter, namely the ``sys.abiflags`` attribute.
+This attribute is a string that contains a number of flags, such as ``d`` for
+debug builds, but it is not easily understood and not uniformly available.
+The new ``sys.abi_features`` attribute is both and is designed to be a superset
+of the information contained in ``sys.abiflags``, which is why we chose to
+include the ``debug`` feature.
```

Comment:
'is both' what? I think it would be better to describe this as "there is already a mechanism X, we introduce a new attribute Y, Y can be used for everything X did and also new capability Z". Currently, the introduction of the attribute is not very well described (e.g. this section doesn't mention it is a frozenset, which attributes will it contain, what format (integer with bitflags, set of strings, enum), etc).